File: unsupported/juno/crates/juno/src/sema/keywords.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

use juno_support::atom_table::Atom;
use juno_support::atom_table::AtomTable;

pub(super) struct Keywords {
    pub ident_arguments: Atom,
    pub ident_eval: Atom,
    pub ident_let: Atom,
    pub ident_new: Atom,
    pub ident_target: Atom,
    pub ident_require: Atom,
}

impl Keywords {
    pub fn new(tab: &AtomTable) -> Self {
        Keywords {
            ident_arguments: tab.atom("arguments"),
            ident_eval: tab.atom("eval"),
            ident_let: tab.atom("let"),
            ident_new: tab.atom("new"),
            ident_target: tab.atom("target"),
            ident_require: tab.atom("require"),
        }
    }
}

File: lib/AST/Keywords.cpp
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#include "hermes/AST/Keywords.h"

#include "hermes/AST/Context.h"
#include "hermes/Support/RegExpSerialization.h"

namespace hermes {
namespace sem {

Keywords::Keywords(Context &astContext)
    : identArguments(
          astContext.getIdentifier("arguments").getUnderlyingPointer()),
      identEval(astContext.getIdentifier("eval").getUnderlyingPointer()),
      identDelete(astContext.getIdentifier("delete").getUnderlyingPointer()),
      identThis(astContext.getIdentifier("this").getUnderlyingPointer()),
      identUseStrict(
          astContext.getIdentifier("use strict").getUnderlyingPointer()),
      identShowSource(
          astContext.getIdentifier("show source").getUnderlyingPointer()),
      identHideSource(
          astContext.getIdentifier("hide source").getUnderlyingPointer()),
      identSensitive(
          astContext.getIdentifier("sensitive").getUnderlyingPointer()),
      identVar(astContext.getIdentifier("var").getUnderlyingPointer()),
      identLet(astContext.getIdentifier("let").getUnderlyingPointer()),
      identConst(astContext.getIdentifier("const").getUnderlyingPointer()),
      identPlus(astContext.getIdentifier("+").getUnderlyingPointer()),
      identMinus(astContext.getIdentifier("-").getUnderlyingPointer()),
      identAssign(astContext.getIdentifier("=").getUnderlyingPointer()),
      identNew(astContext.getIdentifier("new").getUnderlyingPointer()),
      identTarget(astContext.getIdentifier("target").getUnderlyingPointer()),
      identTypeof(astContext.getIdentifier("typeof").getUnderlyingPointer()),
      identConstructor(
          astContext.getIdentifier("constructor").getUnderlyingPointer()),
      identLength(astContext.getIdentifier("length").getUnderlyingPointer()) {}

} // namespace sem
} // namespace hermes
